vertical-pod-autoscaler/pkg/admission-controller/logic/limitrange_checker.go

```
@@ -0,0 +1,251 @@
+					return &neverNeedsLimitsChecker{}
+				}
+			}
+		}
+		return &limitsChecker{limitrangeLister}
+	}
+	return &neverNeedsLimitsChecker{}
+}
+
+type interestingData struct {
+	// Min                  v1.ResourceList
+	// Max                  v1.ResourceList
+	MaxLimitRequestRatio v1.ResourceList
+	Default              v1.ResourceList
+}
+
+func (id *interestingData) parse(lri *v1.LimitRangeItem) {
+	// if value, hasCPU := lri.Max[v1.ResourceCPU]; hasCPU {
+	// }
+	// if value, hasMemory := lri.Max[v1.ResourceMemory]; hasMemory {
+	// }
+	if value, hasCPU := lri.MaxLimitRequestRatio[v1.ResourceCPU]; hasCPU {
+		if id.MaxLimitRequestRatio == nil {
+			id.MaxLimitRequestRatio = make(v1.ResourceList)
+		}
+		if maxRatio, exists := id.MaxLimitRequestRatio[v1.ResourceCPU]; !exists || maxRatio.Cmp(value) > 0 {
+			id.MaxLimitRequestRatio[v1.ResourceCPU] = *value.Copy()
+		}
+	}
+	if value, hasMemory := lri.MaxLimitRequestRatio[v1.ResourceMemory]; hasMemory {
+		if id.MaxLimitRequestRatio == nil {
+			id.MaxLimitRequestRatio = make(v1.ResourceList)
+		}
+		if maxRatio, exists := id.MaxLimitRequestRatio[v1.ResourceMemory]; !exists || maxRatio.Cmp(value) > 0 {
+			id.MaxLimitRequestRatio[v1.ResourceMemory] = *value.Copy()
+		}
+	}
+	if value, hasCPU := lri.Default[v1.ResourceCPU]; hasCPU {
+		if id.Default == nil {
+			id.Default = make(v1.ResourceList)
+		}
+		if _, exists := id.Default[v1.ResourceCPU]; !exists {
+			id.Default[v1.ResourceCPU] = *value.Copy()
+		}
+	}
+	if value, hasMemory := lri.Default[v1.ResourceMemory]; hasMemory {
+		if id.Default == nil {
+			id.Default = make(v1.ResourceList)
+		}
+		if _, exists := id.Default[v1.ResourceMemory]; !exists {
+			id.Default[v1.ResourceMemory] = *value.Copy()
+		}
+	}
+}
+
+func (lc *limitsChecker) getLimitRangeItem(pod *v1.Pod) (ret *v1.LimitRangeItem) {
+	ret = nil
+	limitranges, err := lc.limitrangeLister.
+		LimitRanges(pod.GetNamespace()).
+		List(labels.Everything())
+	if err == nil {
```

I'd rather fail fast:
if err != nil {
  return nil
}

---

Also, it would be good to log the error.